### PR TITLE
debian-patches.sh: clean up maintainer script

### PR DESCRIPTION
**Pull request by butterflya**
Set correct interpreter, fix quoting

###### Things done

- [ ] Tested using sandboxing
  ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS,
    or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file)
    on non-NixOS)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [ ] Linux
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

**Comment by Mic92** [comment]
Looks like the tracker url was replaced by: https://tracker.debian.org/pkg/patch

File: maintainers/scripts/debian-patches.sh [hunk 1]
```diff
@@ -1,17 +1,17 @@
-#!/bin/sh
+#!/usr/bin/env bash
```
Comment (7c6f434c) [inline maintainers/scripts/debian-patches.sh]: I think right now our setup is to have mandatory `/bin/sh` but optional `/ur/bin/env`

File: maintainers/scripts/debian-patches.sh [hunk 2]
```diff
@@ -22,11 +22,11 @@ in
 [
 EOF
 for ((i=1;i < ${#deb_patches[@]}; ++i)); do
-    url="${prefix}/${deb_patches[$i]}"
-    sha256=$(nix-prefetch-url $url)
-    echo "  {"
+    url=${prefix}/${deb_patches[$i]}
```
Comment (7c6f434c) [inline maintainers/scripts/debian-patches.sh]: Why unquote here?

**Review by 7c6f434c — CHANGES_REQUESTED** [review]

File: maintainers/scripts/debian-patches.sh (hunk 1 above)
Comment (butterflya) [inline maintainers/scripts/debian-patches.sh]: The shell script uses features outside of POSIX sh. Specifically, these are bash features.

File: maintainers/scripts/debian-patches.sh (hunk 2 above)
Comment (butterflya) [inline maintainers/scripts/debian-patches.sh]: Because the quotes are redundant.

File: maintainers/scripts/debian-patches.sh [hunk 3]
```diff
@@ -22,11 +22,11 @@ in
...
+    sha256=$(nix-prefetch-url "$url") &&
+    echo '  {'
     echo "    url = \"\${prefix}/${deb_patches[$i]}\";"
     echo "    sha256 = \"$sha256\";"
-    echo "  }"
+    echo '  }'
```
Comment (7c6f434c) [inline maintainers/scripts/debian-patches.sh]: Why would you want to change the quote type here?

Comment (butterflya) [inline maintainers/scripts/debian-patches.sh]: Because it's a constant.

**Comment by 7c6f434c** [comment]
I am not a fitting reviewer for these changes, I find the result harder to read. Some non-purely-cosmetical catches are useful, though.

Convenient GitHub interface doesn't let me to dismiss my change request as obsolete…

**Review by 0xABAB — APPROVED** [review]

**Pull request closed without merging**